tests/helper/test_helper.py

```diff
@@ -177,14 +177,17 @@ def test_powerset(iterable, map_func, expected_powerset):
 )
 def test_classify_unread_counts(mocker, initial_data, stream_dict,
                                 classified_unread_counts, muted_topics,
-                                muted_streams, vary_in_unreads):
+                                muted_streams, vary_in_unreads, initial_index,
+                                index_unread_message_ids):
     model = mocker.Mock()
     model.stream_dict = stream_dict
     model.initial_data = initial_data
     model.muted_topics = muted_topics
     model.muted_streams = muted_streams
-    assert classify_unread_counts(model) == dict(classified_unread_counts,
-                                                 **vary_in_unreads)
+    model.index = initial_index
+    assert classify_unread_counts(model) == dict(
+            classified_unread_counts, **vary_in_unreads)
+    assert model.index['unread_msg_ids'] == index_unread_message_ids
```

Comment:
If `classify_unread_counts` returns a tuple as I mention elsewhere, then the function can be tested without reference to the index, which is simpler.